### DISCUSSION TITLE
How to navigate between pages

### DISCUSSION THREAD

**Discussion by Ohior** [CommunityToolkit/Maui.Markup#245] [Q&A]
I am developing an app that requires multiple pages, and I need to navigate between them and sometimes pass data between them. Many of the example I have seen requires using XAML, and none of them is 100% markup and code base.

**Comment by TheCodeTraveler** [comment]
We have an example included in our sample app.

You'll use [`IQueryAttributable`](https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/shell/navigation#pass-and-process-multiple-items-of-data) on the ViewModel into which you're passing data. And you'll pass in the data as a parameter in `Shell.GoToAsync()`.

https://github.com/CommunityToolkit/Maui.Markup/blob/7327292a8336bb4bbd19d13b23f7f27b8acb4efb/samples/CommunityToolkit.Maui.Markup.Sample/Pages/NewsPage.cs#L91-L104

https://github.com/CommunityToolkit/Maui.Markup/blob/7327292a8336bb4bbd19d13b23f7f27b8acb4efb/samples/CommunityToolkit.Maui.Markup.Sample/ViewModels/NewsDetailViewModel.cs#L34-L43
